recipes/cassandra-cpp-driver/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+        "fPIC": True,
+        "install_header_in_subdir": False,
+        "use_atomic": None,
+        "with_openssl": True,
+        "with_zlib": True,
+        "with_kerberos": False,
+        "use_timerfd": True
+    }
+
+    generators = "cmake"
+    exports_sources = [
+        "CMakeLists.txt",
+        "patches/*"
+    ]
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["VERSION"] = self.version
+        cmake.definitions["CASS_BUILD_EXAMPLES"] = False
+        cmake.definitions["CASS_BUILD_INTEGRATION_TESTS"] = False
+        cmake.definitions["CASS_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["CASS_BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["CASS_BUILD_TESTS"] = False
+        cmake.definitions["CASS_BUILD_UNIT_TESTS"] = False
+        cmake.definitions["CASS_DEBUG_CUSTOM_ALLOC"] = False
+        cmake.definitions["CASS_INSTALL_HEADER_IN_SUBDIR"] = self.options.install_header_in_subdir
+        cmake.definitions["CASS_INSTALL_PKG_CONFIG"] = False
+
+        if self.options.use_atomic == "boost":
+            # Compilation error on Linux
+            if self.settings.os == "Linux":
+                raise ConanInvalidConfiguration(
+                    "Boost.Atomic is not supported on Linux at the moment")
+                    
+            cmake.definitions["CASS_USE_BOOST_ATOMIC"] = True
+            cmake.definitions["CASS_USE_STD_ATOMIC"] = False
+            
+        elif self.options.use_atomic == "std":
+            cmake.definitions["CASS_USE_BOOST_ATOMIC"] = False
+            cmake.definitions["CASS_USE_STD_ATOMIC"] = True
+        else:
+            cmake.definitions["CASS_USE_BOOST_ATOMIC"] = False
+            cmake.definitions["CASS_USE_STD_ATOMIC"] = False
+
+        cmake.definitions["CASS_USE_OPENSSL"] = self.options.with_openssl
+        cmake.definitions["CASS_USE_STATIC_LIBS"] = False
+        cmake.definitions["CASS_USE_ZLIB"] = self.options.with_zlib
+        cmake.definitions["CASS_USE_LIBSSH2"] = False
+
+        if self.options.with_kerberos:
+            # cmake.definitions["CASS_USE_KERBEROS"] = self.options.with_kerberos
+            raise ConanInvalidConfiguration(
+                "Kerberos is not supported at the moment")
+
+        if self.settings.os == "Linux":
+            cmake.definitions["CASS_USE_TIMERFD"] = self.options.use_timerfd
+
+        cmake.configure()
+        return cmake
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.use_timerfd
+
+    def configure(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos is unsupported")
+
+    def requirements(self):
+        self.requires("libuv/1.34.2")
+        self.requires("http_parser/2.9.2")
+        self.requires("rapidjson/cci.20200410")
```

Comment:
we should use the official release over and let consumers can always override

Reply:
@prince-chrismc  Seems that there was a breaking change between `1.1.0` and `cci.20200410`, that changed the `RAPIDJSON_NEW` macro API: https://github.com/Tencent/rapidjson/commit/41ceb8624f2fb46fe5e62b7d2cce471c17db7a5f . And it looks like [library code relies on the new API](https://github.com/datastax/cpp-driver/blob/2.15.3/src/json.hpp#L48) and will not compile with `rapidjson/1.1.0`.

Reply:
Part of the reason that newer one exists! Thanks for confirming that!